SEMANTIC-VERSIONING.md

```diff
@@ -0,0 +1,137 @@
+#Semantic Versioning
+
+When a new version of Tilezen is released, developers should be able to tell from the version increment how much effort it will take them to integrate the new tiles with their map. We use semantic versioning to communicate this.
+
+###What is Semantic Versioning?
+
+Semantic versioning (or [SemVer](http://semver.org/)) is a formalized way of making promises with an X.Y.Z version indicator.
+
+#### Version components
+
+- `MAJOR`.`MINOR`.`PATCH`, example: `1.0.0` (default)
+- `MAJOR`.`MINOR`.`PATCH`-`POSTFIX`, example: `1.0.0-pre1` (optional)
+
+#### Version parts:
+
+1. **MAJOR** version **X** for incompatible API changes.
+2. **MINOR** version **Y** when adding functionality in a backwards-compatible manner, and
+3. **PATCH** version **Z** when fixing backwards-compatible bugs
+4. **pre-release** version **-POSTFIX** when releasing developer previews
+
+**NOTE:** The above applies _after_ software has reached as version `1.0.0`, no promises are made for earlier versions.
+
+#### Developer level of effort:
+
+- Major version X: **high** – significant integration challenges, read the changelog closely
+- Minor version Y: **low** – some integration challenges, read the changelog
+- Bug fixes Z: **none** – simply use the new tiles, ignore the changelog
+- Pre-release POSTFIX: **low to high** – some to significant integration challenges, read the changelog
+
+## Versioning the Tilezen API
+
+Proposed that upon our `1.0.0` release Tilezen makes the following promises.
+
+####MAJOR version increments:
+
+1. **Remove** `common layer`
+1. **Change** `common layer` **name**
+1. **Remove** `common-optional layer`
+1. **Change** `common-optional layer` **name**
+1. **Remove** `common` feature **property** completely from all zooms
+1. **Change** `common` feature **property name**
+1. **Remove** `common-optional` feature **property** at zoom 14 or more
+1. **Change** `common-optional` feature **property name**
+1. **Remove** `optional` feature **property** at zoom 14 or more
+1. **Change** `kind` **value name**
+1. **Remove** `kind` **value** completely from all zooms
+1. **Move** `kind` from one layer to another
+1. **Additional merging** across `kind` **values** in zooms 14, 15, or 16 (or max zoom) by removing `common`, `common-optional`, and/or `optional` **properties** or other method
+1. **Merging** within `kind` **values** at zooms 16 (or max zoom) by removing `common`, `common-optional`, and `optional` **properties** or other method
+1. **Change** of <= -3 (earlier) to default `min_zoom` or `max_zoom` **values** to determine when `kind` is included
+1. **Change** of >= +2 (later) to default `min_zoom` or `max_zoom` **values** to determine when `kind` is included
```

Comment:
Does it make sense to also add in a line for changing the zoom level where we include all data for "overzoom" purposes? Or is that just implied by following the min_zoom values in stylesheets and things would transparently work?